modules/react/avatar/lib/Avatar.tsx

```diff
@@ -1,149 +1,215 @@
 import React, {useState} from 'react';
 import {Property} from 'csstype';
-import {styled, focusRing, hideMouseFocus, filterOutProps} from '@workday/canvas-kit-react/common';
-import isPropValid from '@emotion/is-prop-valid';
-import {borderRadius, colors} from '@workday/canvas-kit-react/tokens';
-import {SystemIconCircle, SystemIconCircleSize} from '@workday/canvas-kit-react/icon';
-import {userIcon} from '@workday/canvas-system-icons-web';
+import {createComponent, focusRing} from '@workday/canvas-kit-react/common';
+import {cssVar, createStencil, calc} from '@workday/canvas-kit-styling';
+import {mergeStyles} from '@workday/canvas-kit-react/layout';
+import {borderRadius} from '@workday/canvas-kit-react/tokens';
+import {SystemIcon, systemIconStencil} from '@workday/canvas-kit-react/icon';
 
-export enum AvatarVariant {
-  Light,
-  Dark,
-}
+import {userIcon} from '@workday/canvas-system-icons-web';
+import {system} from '@workday/canvas-tokens-web';
 
-export interface AvatarProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+export interface AvatarProps {
   /**
-   * The variant of the Avatar default state. Accepts `Light` or `Dark`.
-   * @default AvatarVariant.Light
+   * The variant of the avatar if using a default image. Accepts "light" or "dark."
+   * @default "light"
    */
-  variant?: AvatarVariant;
+  variant?: 'light' | 'dark';
   /**
    * The size of the Avatar.
-   * @default SystemIconCircleSize.m
+   * @default "medium"
    */
-  size?: SystemIconCircleSize | number;
+  size?:
+    | 'extraSmall'
+    | 'small'
+    | 'medium'
+    | 'large'
+    | 'extraLarge'
+    | 'extraExtraLarge'
+    | (string & {});
   /**
-   * The alt text of the Avatar image. This prop is also used for the aria-label
+   * The alt text of the Avatar image. This prop is also used for the aria-label.
    * @default Avatar
    */
   altText?: string;
   /**
-   * The url of the Avatar image.
+   * The URL of the user's photo. Expects a square (1:1) photo
    */
   url?: string;
   /**
-   * The alternative container type for the button. Uses Emotion's special `as` prop.
-   * Will render an `div` tag instead of a `button` when defined.
+   * An onClick function to pass to the underlying `<button>` element.
    */
-  as?: 'div';
+  onClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;
```

Comment:
`onClick` is not needed anymore, since that comes from when the avatar is a button.